src/krb5_plugin/idp/idp_kdcpreauth.c

```diff
@@ -232,6 +232,39 @@ sss_idpkdc_get_complete_attr(const krad_packet *rres,
     return 0;
 }
 
+#ifndef UCHAR_MAX
+#define UCHAR_MAX 255
+#endif
+
+/* RFC 2865 */
+#define MAX_ATTRSIZE (UCHAR_MAX - 2)
+
+#define min(a,b) ((a) > (b) ? (b) : (a))
+static krb5_error_code
+sss_idpkdc_put_complete_attr(krad_attrset *attrset,
+                             krad_attr attr,
+                             const krb5_data *datap)
+{
+    krb5_data state = {0};
+    char *p = datap->data;
+    unsigned int len = datap->length;
+    krb5_error_code ret = 0;
+
+    do {
+        state.data = p;
+        state.length = min(MAX_ATTRSIZE - 5, len);
```

Comment:
@abbra What's the purpose of `- 5` here?

Reply:
It is arbitrary but we need to take a value in a such way that minimal size of a splitted attribute value was more than 3 characters -- there are some requirements in RADIUS specs for attrs. As I said, I am still investigating why krad does not like to parse the attrset generated by this code.

Reply:
@abbra, maybe you've hit https://github.com/krb5/krb5/pull/1229 ?

Reply:
Good question. I created a reproducer but it does not show the problem on f36 beta, which does not have this fix. I am going to rebuild krb5 on my 'faulty' system.

Reply:
Good news. It looks like it was indeed the issue. I've got past that part. Things don't work for me yet for both 'short' and 'long' responses but it is unrelated to this patchset.

Reply:
I will update Fedora krb5 builds with this patch included.

Reply:
I submitted builds to Fedora and will do a series of bodhi updates later tonight.
You'd be able to see them through https://bugzilla.redhat.com/show_bug.cgi?id=2072059

Other than that, I think we can merge this? I pushed FreeIPA updates to my tree and tested them against Microsoft Azure AD configuraitons and also against Google.